packages/world/src/modules/keyswithtable/KeysWithTableModule.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+
+import { IBaseWorld } from "../../interfaces/IBaseWorld.sol";
+import { IModule } from "../../interfaces/IModule.sol";
+
+import { WorldContext } from "../../WorldContext.sol";
+import { ResourceSelector } from "../../ResourceSelector.sol";
+
+import { MODULE_NAMESPACE, USED_KEYS_NAMESPACE } from "./constants.sol";
+import { KeysWithTableHook } from "./KeysWithTableHook.sol";
+import { KeysWithTable } from "./tables/KeysWithTable.sol";
+import { UsedKeysIndex } from "./tables/UsedKeysIndex.sol";
+import { getTargetTableSelector } from "../utils/getTargetTableSelector.sol";
+
+/**
+ * This module deploys a hook that is called when a value is set in the `sourceTableId`
+ * provided in the install methods arguments. The hook keeps track of the keys that are in a given table.
+ * This mapping is stored in a table registered by the module at the `targetTableId` provided in the
+ * install methods arguments.
+ *
+ * Note: if a table with composite keys is used, only the first key is indexed
+ *
+ * Note: this module currently expects to be `delegatecalled` via World.installRootModule.
+ * Support for installing it via `World.installModule` depends on `World.callFrom` being implemented.
+ */
+contract KeysWithTableModule is IModule, WorldContext {
```

Comment:
I realize `KeysWith` is probably trying to align naming with `KeysWithValue` but I feel like it's a little wonky for a table (since a table is "higher level" than keys/values).

Some other name ideas:
- `KeysInTable`
- `KeysForTable`
- `TableKeys`